src/inc/palclr.h

```diff
@@ -484,7 +484,14 @@
 //  and it provides __declspec(selectany) to instruct the linker to merge
 //  duplicate external const static data copies into one.
 //  
+#if defined(__clang__)
+#define SELECTANY extern
```

Comment:
@janvorli The overarching motivation is to get clang to compile the code on Windows so clang-format/-tidy can run on Windows.  It's possible we may be able to tunnel the right compatibility flags through to Clang to get this to be accepted, but I believe we've tried that and failed.  If we can't, we need some workaround here.  In any case, these workarounds' conditions should only trigger in the problem formatting scenarios.  What is the "right" test to detect a Windows build?